doc/source/whatsnew/v2.0.0.rst

```diff
@@ -619,6 +619,7 @@ Other API changes
   new DataFrame (shallow copy) instead of the original DataFrame, consistent with other
   methods to get a full slice (for example ``df.loc[:]`` or ``df[:]``) (:issue:`49469`)
 - Disallow computing ``cumprod`` for :class:`Timedelta` object; previously this returned incorrect values (:issue:`50246`)
+- Instantiating an :class:`Index` with an numeric numpy dtype with data containing :class:`NA` and/or :class:`NaT` now raises a ``ValueError``. Previously a ``TypeError`` was raised (:issue:`51050`)
```

Comment:
does this match what we raise with Series or DataFrame?

Reply:
Yes, so this is a inification wrt. the errors messages the users get, when they do this kind of mistake.